### PR TITLE
[git-username] Bugfix: Allow multiple args

### PR DESCRIPTION
**Pull request by joshuaharry**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
See https://github.com/jonschlinkert/git-username/blob/3baff113f79d62bc779f540ae5f8dfc825dd1fa7/index.js#L13 and https://github.com/jonschlinkert/git-username/blob/3baff113f79d62bc779f540ae5f8dfc825dd1fa7/test.js#L28 and

### CODE REVIEW

File: types/git-username/git-username-tests.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ import username = require('git-username');
 
 username('some/path'); // $ExpectType string | null
 username({ cwd: 'some/path' }); // $ExpectType string | null
-username({ strict: true }); // $ExpectType string
+username({ strict: true }); // $ExpectType string | null
```
Comment (BendingBender) [inline types/git-username/git-username-tests.ts:5]: This test is not correct, when `strict` is set to `true` the function never returns `null`:
```suggestion
username({ strict: true }); // $ExpectType string
```

File: types/git-username/index.d.ts [hunk 2]
```diff
@@ -5,8 +5,10 @@
 
 export = gitUsername;
 
+declare function gitUsername(options?: gitUsername.BaseOptions): string | null;
 declare function gitUsername(options: gitUsername.StrictOptions): string;
-declare function gitUsername(options?: string | gitUsername.BaseOptions): string | null;
+declare function gitUsername(cwd: string, options?: gitUsername.BaseOptions): string | null;
+declare function gitUsername(cwd: string, options?: gitUsername.StrictOptions): string;
```
Comment (BendingBender) [inline types/git-username/index.d.ts:11]: The correct typings for this should never return `null` when `strict` is set to `true`. To achieve this, the typings should look something like the following:

```suggestion
declare function gitUsername(options?: gitUsername.BaseOptions): string | null;
declare function gitUsername(cwd: string): string | null;
declare function gitUsername(cwd: string, options: gitUsername.StrictOptions): string;
// tslint:disable-next-line unified-signatures
declare function gitUsername(cwd: string, options: gitUsername.BaseOptions): string | null;
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/git-username/index.d.ts (hunk 2 above)
Comment (BendingBender) [inline types/git-username/index.d.ts:11]: I think we can get rid of all of this `string | null` debacle. This function can always return a `null` independent of the value of the `strict` option. So this file should actually look something like this:

```ts
declare function gitUsername(cwd?: string, options?: gitUsername.BasicOptions): string | null;
declare function gitUsername(options: gitUsername.FullOptions): string | null;

declare namespace gitUsername {
    interface BasicOptions {
        strict?: boolean;
    }

    interface FullOptions extends BasicOptions {
        cwd?: string;
    }
}
```

File: types/git-username/git-username-tests.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ import username = require('git-username');
...
+username('some-path', { strict: true }); // $ExpectType string | null
```
Comment (BendingBender) [inline types/git-username/git-username-tests.ts:6]: Similarly to the changes in the `index.ts`, the tests should look something like this:
```ts
import username = require('git-username');

// test type exports
type BasicOptions = username.BasicOptions;
type FullOptions = username.FullOptions;

username(); // $ExpectType string | null
username('some/path'); // $ExpectType string | null
username({ cwd: 'some/path' }); // $ExpectType string | null
username({ strict: true }); // $ExpectType string | null
username('some-path', { strict: true }); // $ExpectType string | null
username('some-path', { cwd: 'some/path' }); // $ExpectError
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Comment by joshuaharry** [comment]
@BendingBender Thank you for the suggestions!

**Review by BendingBender — APPROVED** [review]

**Comment by joshuaharry** [comment]
Ready to merge

**Pull request merged**